### ISSUE TITLE
The text of a button is not read by JAWS when referenced with label

### ISSUE THREAD

**Issue by ch0k1** [FreedomScientific/VFO-standards-support#24]
### Summary

When a button element is referenced with a label (either by label's for or button's aria-labelledby). The text of a button is not announced by JAWS.
 
Example:
1. Go to https://jsfiddle.net/ovy0t0ky/2/
2. You will see all the use cases that this issue describes:

* Case 1 - button only
* Case 2 - label + button (button is referenced to the label via label's for attribute)
* Case 3 - label + button (button is referenced  to the label via its aria-labelledby attribute)
* Case 4 - label + button (button is referenced both via label's for attribute and its aria-labelledby attribute)

### Expected result

* Case 1 - The role and text of the button to be announced by JAWS
* Case 2 - The text of the referenced label + button's text and role to be announced by JAWS
* Case 3 - The text of the referenced label + button's text and role to be announced by JAWS
* Case 4 - The text of the referenced label + button's text and role to be announced by JAWS

### Actual result

* Case 1 - The button's text is announced but its role is always omitted by JAWS.
* Case 2 - The text of the referenced label is announced but button's text and role are omitted by JAWS
* Case 3 - The text of the referenced label is announced but button's text and role are omitted by JAWS
* Case 4 - The text of the referenced label is announced but button's text and role are omitted by JAWS

#### JAWS version and build number
JAWS 18.0.4350.400

 
#### Operating System and version
Windows 10 Enterprise
v. 1607 (OS build: 14393.1884)

#### Browser and version:
IE 11.1884.14393.0
Update version 11.0.48 (KB4047206)

**Comment by stevefaulkner** [comment]
thanks for the report @ch0k1, will look into it. btw have created a codepen https://codepen.io/stevef/pen/yPxJmm as i find it easier to use.

**Comment by stevefaulkner** [comment]
@ch0k1 how are you navigating to the button when jaws used?

**Comment by stevefaulkner** [comment]
@ch0k1 in absence of understanding how you are navigating, will provide feedback:
I tested [your demo](https://jsfiddle.net/ovy0t0ky/2/) in JAWS 18.0.453.0 and IE11 focusing each button either using the `b` key or the `tab` key:

* Case 1 - button only - **works as expected button label and role announced**
* Case 2 - label + button (button is referenced to the label via label's for attribute) - `works as expected the `<label>` content and role is announced. The child text of the `button` is not announced as it is overridden by the `<label>`.`
* Case 3 - label + button (button is referenced  to the label via its aria-labelledby attribute)  - **works as expected the `<label>` content and role is announced. The child text of the `button` is not announced as it is overridden by the `<label>`.**
* Case 4 - label + button (button is referenced both via label's for attribute and its aria-labelledby attribute)  - `works as expected the `<label>` content and role is announced. The child text of the `button` is not announced as it is overridden by the `<label>`.`

I think you misunderstand how concatenation of labels works. In order for the 'double labeled' buttons to have both the `<label>` and the child text of the button used as the accessible name, both have to be explicitly referenced by the `aria-labelledby`, like this:

`<button id="btn3" aria-labelledby="lbl2 innerlabel3"><span id="innerlabel3" tabindex="-1">Double labelled button</span></button>`

**note** also that due to a [bug in IE](https://w3c.github.io/using-aria/#label-support), the inner `span` requires `tabindex="-1"`
Here is a [codepen](https://codepen.io/stevef/pen/yPxJmm) that demonstrates the markup requirements to get the expected behavior.

**Comment by NikolayCholakov** [comment]
Hello @stevefaulkner,

Sorry for the delayed answer from my side - I access the buttons via the `tab` key.

Thanks for your fast answer and hints for how to accomplish the 'double labeled' behavior

Best regards,
@ch0k1

**Comment by stevefaulkner** [comment]
@ch0k1 no worries, hope the suggested solution resolves your issues.

**Comment by stevefaulkner** [comment]
@ch0k1 no worries, hope the suggested solution resolves your issues.

**Comment by ggordon-vfo** [comment]
Doesn't appear that anything needs changing on the JAWS side; closing the issue.
